docs/Running-Tachyon-Fault-Tolerant-on-EC2.md

```diff
@@ -4,4 +4,143 @@ title: Running Tachyon with Fault Tolerance on EC2
 nickname: Tachyon on EC2 with Fault Tolerance
 group: User Guide
 priority: 3
----
\ No newline at end of file
+---
+
+Tachyon with Fault Tolerance can be deployed on EC2 using the [Vagrant scripts](https://github.com/amplab/tachyon/tree/master/deploy/vagrant) that come with Tachyon. The scripts let you create, configure and destroy clusters that come automatically configured with HDFS.
+
+# Prerequisites
+
+**Install Vagrant and the AWS plugins**
+
+Download [Vagrant](https://www.vagrantup.com/downloads.html)
+
+Install AWS Vagrant plugin:
+
+    vagrant plugin install vagrant-aws
+    vagrant box add dummy https://github.com/mitchellh/vagrant-aws/raw/master/dummy.box
+
+**Install Tachyon**
+
+Download Tachyon to your local machine, and unzip it:
+
+    $ wget https://github.com/amplab/tachyon/releases/download/v{{site.TACHYON_RELEASED_VERSION}}/tachyon-{{site.TACHYON_RELEASED_VERSION}}-bin.tar.gz
+    $ tar xvfz tachyon-{{site.TACHYON_RELEASED_VERSION}}-bin.tar.gz
+
+**Install python library dependencies**
+
+Install [python>=2.7](https://www.python.org/), not python3.
+
+Under `deploy/vagrant` directory in your home directory, run:
+
+    $ sudo bash bin/install.sh
+
+Alternatively, you can manually install [pip](https://pip.pypa.io/en/latest/installing/), and then in `deploy/vagrant` run:
+
+    $ sudo pip install -r pip-req.txt
+
+# Launch a Cluster
+
+To run a Tachyon cluster on EC2, first sign up for an Amazon EC2 account
+on the [Amazon Web Services site](http://aws.amazon.com/). 
+
+Then create [access keys](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html) 
+and set shell environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` by:
+
+    $ export AWS_ACCESS_KEY_ID=<your access key>
+    $ export AWS_SECRET_ACCESS_KEY=<your secret access key>
+
+Next generate your EC2 [Key Pairs](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html). Make sure to set the permissions of your private key file that only you can read it:
+
+    $ chmod 400 <your key pair>.pem
+
+In the configuration file `deploy/vagrant/conf/ec2.yml`, set the value of `Keypair` to your keypair name and `Key_Path` to the path to the pem key.
+
+In the configuration file `deploy/vagrant/conf/tachyon.yml`, set the value of `Masters` to the number of TachyonMasters you want.
```

Comment:
Here you should explain that you need more than 1 master for fault tolerant mode.